src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/UsePropertiesWhereAppropriateTests.cs

```diff
@@ -1,23 +1,415 @@
+
+    public string Get123()
+    {
+        return fileName;
+    }|]
+
+    protected string GetFileNameProtected()
+    {
+        return fileName;
+    }
+}
+",
+            GetCA1024CSharpResultAt(11, 19, "Get_FileName2"),
+            GetCA1024CSharpResultAt(16, 19, "Get123"));
+        }
+
+        [Fact]
+        public void VisualBasic_CA1024NoDiagnosticCases()
+        {
+            VerifyBasic(@"
+Public Class Base
+	Public Overridable Function GetSomething() As Integer
+		Return 0
+	End Function
+End Class
+
+Public Class Class1
+	Inherits Base
+	Private fileName As String
+
+	' 1) Returns void
+	Public Sub GetWronglyNamedMethod()
+	End Sub
+
+	' 2) Not a method
+	Public ReadOnly Property LogFile() As String
+		Get
+			Return fileName
+		End Get
+	End Property
+
+	' 3) Returns an array type
+	Public Function GetValues() As Integer()
+		Return Nothing
+	End Function
+
+	' 4) Has parameters
+	Public Function GetMethodWithParameters(p As Integer) As Integer()
+		Return New Integer() {p}
+	End Function
+
+	' 5a) Name doesn't start with a 'Get'
+	Public Function SomeMethod() As Integer
+		Return 0
+	End Function
+
+	' 5b) First compound word is not 'Get'
+	Public Function GetterMethod() As Integer
+		Return 0
+	End Function
+
+	' 6) Generic method
+	Public Function GetGenericMethod(Of T)() As Object
+		Return New GenericType(Of T)()
+	End Function
+
+	' 7) Override
+	Public Overrides Function GetSomething() As Integer
+		Return 1
+	End Function
+
+	' 8) Method with overloads
+	Public Overrides Function GetOverloadedMethod() As Integer
+		Return 1
+	End Function
+
+	Public Overrides Function GetOverloadedMethod(i As Integer) As Integer
+		Return i
+	End Function
+
+	' 9) Methods with special name
+	Public Overloads Function GetHashCode() As Integer
+		Return 0
+	End Function
+
+	Public Function GetEnumerator() As IEnumerator
+		Return Nothing
+	End Function
+
+	' 10) Method with invocation expressions
+	Public Function GetSomethingWithInvocation() As Integer
+		System.Console.WriteLine(Me)
+		Return 0
+	End Function
+
+	' 11) Method named 'Get'
+	Public Function Get() As String
+		Return fileName
+	End Function
+
+    ' 12) Private method
+	Private Function GetSomethingPrivate() As String
+		Return fileName
+	End Function
+
+    ' 13) Friend method
+	Friend Function GetSomethingInternal() As String
+		Return fileName
+	End Function
+End Class
+");
+        }
+
+        [Fact]
+        public void CSharp_CA1024NoDiagnosticOnUnboundMethodCaller()
+        {
+            VerifyCSharp(@"
+public class class1
+{
+    public int GetSomethingWithUnboundInvocation()
+    {
+        Console.WriteLine(this);
+        return 0;
+    }
+}
+");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/7222")]
+        public void VisualBasic_CA1024NoDiagnosticOnUnboundMethodCaller()
+        {
+            VerifyBasic(@"
+Public Class class1
+	Public Function GetSomethingWithUnboundInvocation() As Integer
+		Console.WriteLine(Me)
```

Comment:
In this case, Console.WriteLine is unbound (since a `Imports System` statement is missing) and the VB compiler produces an BadExpression for it instead of an InvocationExpression with a null symbol. C# does the right thing here.